src/scaling.jl

```diff
@@ -12,93 +12,65 @@ Represents the no-op scaling which simply returns the `data` it is applied on.
-
+@inline _apply(x, ::IdentityScaling; kwargs...) = x
+@inline _apply!(x, ::IdentityScaling; kwargs...) = x
 
 """
-    MeanStdScaling(mean, std) <: AbstractScaling
+    MeanStdScaling(μ, σ) <: AbstractScaling
 
-Linearly scale the data by a statistical `mean` and standard deviation `std`.
+Linearly scale the data by the statistical mean `μ` and standard deviation `σ`.
 This is also known as standardization, or the Z score transform.
-Once computed, the statistics of a `MeanStdScaling` are immutable.
-
-Can take a precomputed `mean` and `std` as arguments, or compute them from data.
-
-# Arguments
-* `mean::NamedTuple`: tuple of mean values, named by the scope of values it applies to.
-  `(all=μ, )` will apply to all data; `(1=μ1, 2=μ2)` for `AbstractArray` data will apply μ1
-  to the first slice and μ2 to the second slice; `(a=μ1, b=μ2)` for `Table` data will apply
-  μ1 to column `a` and μ2 to column `b`.
-* `std::NamedTuple`: similar to `mean` but for standard deviation values.
 
 # Keyword arguments to `apply`
 * `inverse=true`: inverts the scaling (e.g. to reconstruct the unscaled data)
 * `eps=1e-3`: replaces all 0 values in `std` before scaling (if `inverse=false`)
 """
 struct MeanStdScaling <: AbstractScaling
-    mean::NamedTuple
-    std::NamedTuple
-end
-
-"""
-    MeanStdScaling(data; kwargs...) <: Scaling
-
-Construct a [`MeanStdScaling`](@ref) using the mean and standard deviation of the data.
-
-!!! note
-    `dims` and `cols` keyword arguments must be specified the same way when constructing
-    and applying the transform.
-    Otherwise, the results will be inconsistent, or an error may occur.
-
-# Keyword arguments
-* `dims=:`: for `AbstractArray` data, the dimension(s) to compute statistics along.
-* `cols=nothing`: for `Table` data, the column names to compute statistics for.
-"""
-function MeanStdScaling(data; kwargs...)
-    μ, σ = compute_stats(data; kwargs...)
-    return MeanStdScaling(μ, σ)
-end
+    μ::Real
+    σ::Real
+
+    """
+        MeanStdScaling(A::AbstractArray; dims=:, inds=:) -> MeanStdScaling
+        MeanStdScaling(table, cols=nothing) -> MeanStdScaling
+
+    Construct a [`MeanStdScaling`](@ref) transform from the statistics of the given data.
+    Default behaviour is to consider all data when computing the mean and standard deviation.
+    This can be restricted to certain slices via the keyword arguments (see below).
+
+    # `AbstractArray` keyword arguments
+    * `dims=:`: the dimension along which to take the `inds` slices.
+    * `inds=:`: the slices of the array to use in computing the statistics. Default uses all slices.
+
+    # `Table` keyword arguments
+    * `cols=nothing`: the columns to use in computing the statistics. Default uses all columns.
+
+    !!! note
+        The keyword arguments must be specified the same way when you construct and [`apply`](@ref)
+        the transform, else the results will be inconsistent, or throw an error.
```

Comment:
So I guess the rationale of still having to specify `dims` and `inds` in `apply` is: we're providing the freedom to scale whatever data/dimension/slice the user wants.

Given that, I think this note should be rephrased to say that if you _want_ it to be consistent, you _should_ specify the same way.

Reply:
reworded the docstring(s)